include/geode/basic/attribute.h

```diff
@@ -130,29 +138,31 @@ namespace geode
 
         virtual float generic_value( index_t element ) const = 0;
 
-    private:
-        template < typename Archive >
-        void serialize( Archive& archive )
-        {
-            archive.ext( *this, DefaultGrowable< Archive, AttributeBase >{},
-                []( Archive& /*unused*/, AttributeBase& /*unused*/ ) {} );
-        }
+        virtual absl::string_view type() = 0;
 
-        virtual void resize( index_t size ) = 0;
+        virtual void resize( index_t size, AttributeKey /*unused*/ ) = 0;
```

Comment:
Why adding the Key for all methods?

Reply:
thats all the methods that only the AttributeManager should be able to call

Reply:
And functions are now in public?

Reply:
yes, public defines the callable methods and private the one not callable. With passkey you can restrict who can can the callables but in a sense there are callable from the outside.